pkg/apis/core/helper/qos/qos.go

```diff
@@ -40,10 +41,7 @@ func GetPodQOS(pod *core.Pod) core.PodQOSClass {
 	limits := core.ResourceList{}
 	zeroQuantity := resource.MustParse("0")
 	isGuaranteed := true
-	allContainers := []core.Container{}
-	allContainers = append(allContainers, pod.Spec.Containers...)
-	allContainers = append(allContainers, pod.Spec.InitContainers...)
-	for _, container := range allContainers {
+	podutil.VisitContainers(&pod.Spec, podutil.Containers|podutil.InitContainers, func(container *core.Container, containerType podutil.ContainerType) bool {
```

Comment:
was this intentionally ignoring ephemeral containers or was that a bug?

Reply:
https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/

> Pod resource allocations are immutable, so setting resources is disallowed.


- https://github.com/kubernetes/kubernetes/pull/115485
- https://github.com/kubernetes/website/pull/39235